src/main/java/inf112/skeleton/app/board/Board.java

```diff
@@ -0,0 +1,194 @@
+package inf112.skeleton.app.board;
+
+import com.badlogic.gdx.graphics.g2d.TextureRegion;
+import com.badlogic.gdx.maps.MapLayers;
+import com.badlogic.gdx.maps.tiled.TiledMapTile;
+import com.badlogic.gdx.maps.tiled.tiles.StaticTiledMapTile;
+import inf112.skeleton.app.object.Flag;
+import com.badlogic.gdx.maps.tiled.TiledMap;
+import com.badlogic.gdx.maps.tiled.TiledMapTileLayer;
+import com.badlogic.gdx.maps.tiled.TmxMapLoader;
+import inf112.skeleton.app.object.Hole;
+import inf112.skeleton.app.object.Robot;
+
+import java.util.ArrayList;
+
+public class Board {
+    public ArrayList<Flag> flags;
+    public ArrayList<Hole> holes;
+
+    public TiledMap map_TiledMap;
+
+    private TiledMapTileLayer board_MapLayer;
+    public TiledMapTileLayer players_MapLayer;
+    private TiledMapTileLayer flags_MapLayer;
+    private TiledMapTileLayer holes_MapLayer;
+
+    private int height;
+    private int width;
+    private int tileHeight;
+
+    /*
+     * construct a game board form the map that is inputted with a filename
+     *
+     * @param fileName a map file
+     */
+    public Board(String fileName){
+        map_TiledMap = new TmxMapLoader().load(fileName);
+
+        height = (int) map_TiledMap.getProperties().get("height");
+        width = (int) map_TiledMap.getProperties().get("width");
+        tileHeight = (int) map_TiledMap.getProperties().get("tileheight");
+
+        board_MapLayer = (TiledMapTileLayer) map_TiledMap.getLayers().get("Board");
+        players_MapLayer = (TiledMapTileLayer) map_TiledMap.getLayers().get("Players");
+        flags_MapLayer = (TiledMapTileLayer) map_TiledMap.getLayers().get("Flags");
+        holes_MapLayer = (TiledMapTileLayer) map_TiledMap.getLayers().get("Holes");
+
+
+
+        flags = new ArrayList<Flag>();
+        holes = new ArrayList<Hole>();
+
+        initFlags();
+        initHoles();
+    }
+
+
+    /*
+     * for tests
```

Comment:
What is this!

Reply:
This is so that we run the test without the tilemap